Kernel/Arch/x86/i386/Processor.cpp

```diff
@@ -75,43 +76,40 @@ FlatPtr Processor::init_context(Thread& thread, bool leave_crit)
         m_in_critical = 1; // leave it without triggering anything or restoring flags
     }
 
-    u32 kernel_stack_top = thread.kernel_stack_top();
+    // TODO: handle NT?
+    VERIFY((cpu_flags() & 0x24000) == 0); // Assume !(NT | VM)
 
-    // Add a random offset between 0-256 (16-byte aligned)
-    kernel_stack_top -= round_up_to_power_of_two(get_fast_random<u8>(), 16);
+    // Get stack top and add a random offset between 0-256 (16-byte aligned)
+    FlatPtr const kernel_stack_top = thread.kernel_stack_top() - round_up_to_power_of_two(get_fast_random<u8>(), 16);
 
-    u32 stack_top = kernel_stack_top;
+    StackWriter w(kernel_stack_top);
```

Comment:
This could use a more descriptive name, how about

```suggestion
    StackWriter kernel_stack_writer(kernel_stack_top);
```

Reply:
I opted for plain stack_writer, hope that's okay.